File: android/live-update-provider/src/main/kotlin/io/ionic/liveupdateprovider/ProviderSyncResult.kt
package io.ionic.liveupdateprovider

/** A provider-specific result from a sync operation. */
interface ProviderSyncResult

/**
 * A sync result containing provider metadata.
 *
 * @param metadata Metadata containing JSON-serializable values.
 */
data class MetadataSyncResult(
    val metadata: Map<String, Any>
) : ProviderSyncResult

File: ios/Sources/LiveUpdateProvider/ProviderSyncResult.swift
/// A provider-specific result from a sync operation.
public protocol ProviderSyncResult {}

/// A sync result containing provider metadata.
public struct MetadataSyncResult: ProviderSyncResult {
    /// Metadata containing JSON-serializable values.
    public let metadata: [String: Any]

    public init(metadata: [String: Any]) {
        self.metadata = metadata
    }
}
